libs/dmlf/include/dmlf/distributed_learning/utilities/mnist_client_utilities.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "dmlf/distributed_learning/distributed_learning_client.hpp"
+#include "ml/dataloaders/mnist_loaders/mnist_loader.hpp"
+#include "ml/ops/activations/relu.hpp"
+#include "ml/ops/loss_functions/cross_entropy_loss.hpp"
+#include "ml/ops/placeholder.hpp"
+#include "ml/optimisation/adam_optimiser.hpp"
+
+namespace fetch {
+namespace dmlf {
+namespace distributed_learning {
+namespace utilities {
+
+template <typename TensorType>
+std::shared_ptr<fetch::dmlf::distributed_learning::TrainingClient<TensorType>> MakeMNISTClient(
```

Comment:
I was imagining a more general MakeClient method for which we specify the architecture of the model. However, it's probably fine to leave it as it is and simply rework it as part of a refactor that lets clients have models instead of separate graph, dataloader, and optimiser